pkg/util/slices.go

```diff
@@ -121,3 +121,14 @@ func Reduce[T any, U any](collection []T, fn func(acc U, el T, idx int) U, init
 	}
 	return acc
 }
+
+// FindIndex returns the index of the first element in the collection that
+// satisfies the predicate. If no element satisfies the predicate, returns -1.
+func FindIndex[T any](collection []T, predicate func(T) bool) int {
```

Comment:
Isn't this the same as https://pkg.go.dev/slices@go1.22.0#IndexFunc?

Reply:
Ah yes... it is.

![](https://media.tenor.com/cHYRjCIXyckAAAAM/why-not-both-por-que-no-los-dos.gif)